sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -436,9 +437,10 @@ class ReceivedMessage(PeekMessage):
             :dedent: 4
             :caption: Checking the properties on a received message.
     """
-    def __init__(self, message, mode=ReceiveSettleMode.PeekLock):
+    def __init__(self, message, mode=ReceiveSettleMode.PeekLock, is_deferred_message=False):
```

Comment:
I'm unsettled by something _so thoroughly internal_ (and single-use) being exposed even this much.  To the extent that part of me is like "well just try settling it via link and fall back if it fails, we'd have that logic anyway for the happy-path fallback scenario"; but that also seems a bit nasty unless the failure is VERY lightweight.

at the very least making it a KW Only arg might help us slightly.  Can simply add a * in between the mode and is_deferred, args after it become non-positional.

Reply:
updated it to kwargs